cart/application/cartService.go

```diff
@@ -861,3 +873,42 @@ func (cs *CartService) dispatchAllEvents(ctx context.Context, events []flamingo.
 		cs.eventRouter.Dispatch(ctx, e)
 	}
 }
+
+// AdjustItemsToRestrictedQty checks the quantity restrictions for each item of the cart and returns what quantities have been adjusted
+func (cs *CartService) AdjustItemsToRestrictedQty(ctx context.Context, session *web.Session) (AdjustmentResults, error) {
+	cart, _, err := cs.cartReceiverService.GetCart(ctx, session)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make(AdjustmentResults, 0)
+
+	for _, delivery := range cart.Deliveries {
+		for _, item := range delivery.Cartitems {
+			product, err := cs.productService.Get(ctx, item.MarketplaceCode)
+			if err != nil {
+				return nil, err
+			}
+
+			restrictionResult := cs.restrictionService.RestrictQty(ctx, product, cart, delivery.DeliveryInfo.Code)
+
+			if restrictionResult.RemainingDifference < 0 {
+				newQty := item.Qty + restrictionResult.RemainingDifference
+
+				err = cs.UpdateItemQty(ctx, session, item.ID, delivery.DeliveryInfo.Code, newQty)
```

Comment:
Es gibt ein "DeleteItem" - wenn die Qty 0 (oder kleiner) wäre sollten wir DeleteItem aufrufen und im Adjustmentresult das "WasDeleted true" setzen)

Reply:
DeleteItem wird in UpdateItemQty aufgerufen, falls qty < 0 ist. Könnten dies aber auch hier abfragen und DeleteItem selbst aufrufen.

Reply:
Implemented as suggested.